include/rapidjson/writer.h

```diff
@@ -150,6 +150,15 @@ class Writer {
         return maxDecimalPlaces_;
     }
 
+    bool DepthinArrayChange() const {
+        static int prev_depth=0;
```

Comment:
Thanks for review, I totally agree! static is removed.
I also agree about the one leaf array but this is not the case right now...  Could learn the maxDepth along the way... I'll give it a try..